.ci/generate_test_report_lib.py

```diff
@@ -112,15 +113,46 @@ def generate_report(
                 ]
             )
         else:
-            report.extend(
-                [
-                    "The build failed before running any tests.",
-                    "",
-                    SEE_BUILD_FILE_STR,
-                    "",
-                    UNRELATED_FAILURES_STR,
-                ]
-            )
+            ninja_failures = find_failure_in_ninja_logs(ninja_logs)
+            if not ninja_failures:
+                report.extend(
+                    [
+                        "The build failed before running any tests. Detailed "
+                        "information about the build failure could not be "
+                        "automatically obtained.",
+                        "",
+                        SEE_BUILD_FILE_STR,
+                        "",
+                        UNRELATED_FAILURES_STR,
+                    ]
+                )
+            else:
+                report.extend(
+                    [
+                        "The build failed before running any tests. Click on the "
+                        "failure below to see the details.",
+                        "",
+                    ]
+                )
+                for build_failure in ninja_failures:
+                    failed_action, failure_message = build_failure
+                    report.extend(
+                        [
+                            "<details>",
+                            f"<summary>{failed_action}</summary>",
+                            "",
+                            "```",
+                            failure_message,
+                            "```",
+                            "</details>",
+                        ]
+                    )
```

Comment:
Could dedupe this part and put it in a utility function "format_ninja_failures", something like that.